File: programmers/js/basic/181925.js
// 수 조작하기 2
function solution(numLog) {
  let answer = "";

  for (let i = 1; i < numLog.length; i++) {
    const calcNum = numLog[i] - numLog[i - 1];

    if (calcNum === 1) answer += "w";
    if (calcNum === -1) answer += "s";
    if (calcNum === 10) answer += "d";
    if (calcNum === -10) answer += "a";
  }

  return answer;
}

File: programmers/java/basic/181925.java
// 수 조작하기 2

import java.util.HashMap;

class Solution {
    public String solution(int[] numLog) {
        String result = "";
        HashMap<Integer, String> map = new HashMap<>(){{
            put(1, "w");
            put(-1, "s");
            put(10, "d");
            put(-10, "a");
        }};
        
        for (int i = 1; i < numLog.length; i++) {
            result += map.get(numLog[i] - numLog[i - 1]);
        }
        
        return result;
    }
}